### PR TITLE
Doc aliases for pre-1.0 function names

### PR DESCRIPTION
**Pull request by kornelski**
To aid migration from 0.x:

Screen Shot 2021-02-14 at 14 35 27

### CODE REVIEW

**Review by taiki-e — COMMENTED** [review]
Thanks for the PR! Could you add comments explaining why these aliases exist? (like https://github.com/tokio-rs/bytes/pull/471 did)

File: tokio/src/io/async_fd.rs [hunk 1]
```diff
@@ -519,6 +519,7 @@ impl<'a, Inner: AsRawFd> AsyncFdReadyGuard<'a, Inner> {
     /// create this `AsyncFdReadyGuard`.
     ///
     /// [`WouldBlock`]: std::io::ErrorKind::WouldBlock
+    #[doc(alias = "with_io")]
```
Comment (taiki-e) [inline tokio/src/io/async_fd.rs:522]:
```suggestion
    #[cfg_attr(docsrs, doc(alias = "with_io"))]
```

File: tokio/src/runtime/builder.rs [hunk 2]
```diff
@@ -233,6 +233,7 @@ impl Builder {
     /// # Panic
     ///
     /// This will panic if `val` is not larger than `0`.
+    #[doc(alias = "max_threads")]
```
Comment (taiki-e) [inline tokio/src/runtime/builder.rs:236]:
```suggestion
    #[cfg_attr(docsrs, doc(alias = "max_threads"))]
```

File: tokio/src/sync/notify.rs [hunk 3]
```diff
@@ -312,6 +312,7 @@ impl Notify {
     ///     notify.notify_one();
     /// }
     /// ```
+    #[doc(alias = "notify")]
```
Comment (taiki-e) [inline tokio/src/sync/notify.rs:315]:
```suggestion
    #[cfg_attr(docsrs, doc(alias = "notify"))]
```

File: tokio/src/time/driver/sleep.rs [hunk 4]
```diff
@@ -16,6 +16,7 @@ use std::task::{self, Poll};
 ///
 /// Canceling a sleep instance is done by dropping the returned future. No additional
 /// cleanup work is required.
+#[doc(alias = "delay_until")]
```
Comment (taiki-e) [inline tokio/src/time/driver/sleep.rs:19]:
```suggestion
#[cfg_attr(docsrs, doc(alias = "delay_until"))]
```

File: tokio/src/time/driver/sleep.rs [hunk 5]
```diff
@@ -53,6 +54,7 @@ pub fn sleep_until(deadline: Instant) -> Sleep {
 /// ```
 ///
 /// [`interval`]: crate::time::interval()
+#[doc(alias = "delay_for")]
```
Comment (taiki-e) [inline tokio/src/time/driver/sleep.rs:57]:
```suggestion
#[cfg_attr(docsrs, doc(alias = "delay_for"))]
```

File: tokio/src/io/async_fd.rs (hunk 1 above)
Comment (kornelski) [reply tokio/src/io/async_fd.rs:522]: I don't see the benefit of disabling it for local `cargo doc`.

Comment (taiki-e) [reply tokio/src/io/async_fd.rs:522]: `doc(alias)` is not stable in tokio's MSRV, so we need to use `doc(alias)` only when cfg for documentation is enabled. (Sorry for forgot to explain why these changes are needed!)
https://github.com/tokio-rs/tokio/runs/1897938802?check_suite_focus=true#step:4:93

Comment (kornelski) [reply tokio/src/io/async_fd.rs]: I see. Thanks

**Review by taiki-e — APPROVED** [review]

File: tokio/src/runtime/builder.rs [hunk 6]
```diff
@@ -233,6 +233,8 @@ impl Builder {
     /// # Panic
     ///
     /// This will panic if `val` is not larger than `0`.
+    // Alias for old name in 0.x
+    #[cfg_attr(docsrs, doc(alias = "max_threads"))]
```
Comment (Darksonn) [inline tokio/src/runtime/builder.rs:237]: It's worth noting that, strictly speaking, `max_blocking_threads` does not do the same as `max_threads`.

Comment (kornelski) [reply tokio/src/runtime/builder.rs:237]: I can't find anything about the semantic change in the changelog or the PR https://github.com/tokio-rs/tokio/pull/3287

Comment (Darksonn) [reply tokio/src/runtime/builder.rs:237]: Previously `max_threads` limited both blocking and worker threads, but the new `max_blocking_thread` does not include async worker threads in the count.

**Comment by kornelski** [comment]
ping

**Review by Darksonn — APPROVED** [review]

**Pull request merged**